### DISCUSSION TITLE
can somebody fix this for me please?

### DISCUSSION THREAD

**Discussion by hammouda101010** [theJayTea/WritingTools#46] [Q&A] [answered]
```json
{
    "shortcut": "ctrl+space",
    "theme": "gradient",
    "providers": {
        "Gemini 1.5 Flash (Recommended)": {},
        "OpenAI Compatible (For Experts)": {
            "api_key": "hidden",
            "api_base": "https://api.groq.io",
            "api_organisation": "Groq",
            "api_project": "",
            "api_model": "llama-3.1-8b-instant"
        }
    },
    "streaming": false,
    "provider": "OpenAI Compatible (For Experts)"
}
```

**Comment by theJayTea** [comment, marked as answer]
Hey @hammouda101010 , you've entered the Grok API URL incorrectly. Please refer to their own documentation on this:

https://console.groq.com/docs/api-reference#chat-create

The API base URL must be: 
https://api.groq.com/openai/v1/chat/completions

I've not gone through their whole documentation page, but from my cursory look, this is one mistake I spotted. Refer to their documentation correctly to fix it. It shouldn't take more than 2 minutes. Cheers!

**Reply by hammouda101010** [reply]
@theJayTea what model can i use with this base url?

**Reply by hammouda101010** [reply]
also, this still doesn't work...

```json
{
    "shortcut": "ctrl+space",
    "theme": "gradient",
    "providers": {
        "Gemini 1.5 Flash (Recommended)": {},
        "OpenAI Compatible (For Experts)": {
            "api_key": "hidden",
            "api_base": "https://api.groq.com/openai/v1/chat/completions",
            "api_organisation": "Groq",
            "api_project": "writing-tools",
            "api_model": "llama3-8b-8192"
        }
    },
    "streaming": false,
    "provider": "OpenAI Compatible (For Experts)"
}
```
also can't get an gemmini API key because it's blocked in my region.

**Reply by theJayTea** [reply]
@hammouda101010  My bad, that documentation I linked previously looks like it’s for Groq’s _own_ API. Check this out for their OpenAI-style API that Writing Tools needs:
https://console.groq.com/docs/openai

So your base URL should be:
https://api.groq.com/openai/v1

And of course, you’ll need to enter your Groq API key as mentioned in this documentation.

Groq might not have a free tier, and you may have to first pay for its credits for it to work, just making sure you know.

BTW, if you have a PC with a GPU, you can also run a local LLM with llama - it works seamlessly and is completely free.

**Reply by hammouda101010** [reply]
> @hammouda101010 My bad, that documentation I linked previously looks like it’s for Groq’s _own_ API. Check this out for their OpenAI-style API that Writing Tools needs: https://console.groq.com/docs/openai
> 
> So your base URL should be: https://api.groq.com/openai/v1
> 
> And of course, you’ll need to enter your Groq API key as mentioned in this documentation.
> 
> Groq might not have a free tier, and you may have to first pay for its credits for it to work, just making sure you know.
> 
> BTW, if you have a PC with a GPU, you can also run a local LLM with llama - it works seamlessly and is completely free.

nah don't worry. it's API is free!
